packages/graphql-testing/src/Requests.ts

```diff
@@ -0,0 +1,51 @@
+import {GraphQLRequest} from 'apollo-link';
+
+export default class Requests {
+  private requests: GraphQLRequest[] = [];
+
+  constructor(requests?: GraphQLRequest[]) {
+    this.requests = requests ? [...requests] : [];
+  }
+
+  get last() {
+    return this.requests[this.requests.length - 1];
+  }
+
+  [Symbol.iterator]() {
+    return this.requests[Symbol.iterator]();
+  }
+
+  nth(index: number) {
+    return index < 0
+      ? this.requests[this.requests.length - 1 + index]
+      : this.requests[index];
+  }
+
+  push(request: GraphQLRequest) {
+    this.requests.push(request);
+  }
+
+  allOperations(): GraphQLRequest[] {
+    return this.requests;
+  }
+
+  allWithOperationName(operation: string): GraphQLRequest[] {
```

Comment:
I think maybe this method should go away in favour of an optional `{name?: string}` option to `allOperations`